pkg/generator/github/github.go

```diff
@@ -84,20 +85,28 @@ func (g *Generator) generate(
 		return nil, fmt.Errorf("error creating request: %w", err)
 	}
 
-	body := make(map[string]interface{})
+	payload := make(map[string]interface{})
 	if gh.Permissions != nil {
-		body["permissions"] = gh.Permissions
+		payload["permissions"] = gh.Permissions
 	}
 	if len(gh.Repositories) > 0 {
-		body["repositories"] = gh.Repositories
+		payload["repositories"] = gh.Repositories
 	}
-	bodyBytes, err := json.Marshal(body)
-	if err != nil {
-		return nil, fmt.Errorf("error creating request: %w", err)
+
+	var body io.Reader
+	if len(payload) > 0 {
+		bodyBytes, err := json.Marshal(payload)
+		if err != nil {
+			return nil, fmt.Errorf("error marshalling payload: %w", err)
+		}
+
+		body = bytes.NewReader(bodyBytes)
+	} else {
+		body = http.NoBody
 	}
```

Comment:
```suggestion
	body := http.NoBody
	if len(payload) > 0 {
		bodyBytes, err := json.Marshal(payload)
		if err != nil {
			return nil, fmt.Errorf("error marshalling payload: %w", err)
		}

		body = bytes.NewReader(bodyBytes)
	}
```

No need for the else.

Reply:
sorry, 
```
cannot use bytes.NewReader(bodyBytes) (value of type *bytes.Reader) as http.noBody value in assignment
```

Reply:
Ah yes. Use this:
```go
	var body io.Reader = http.NoBody
	if len(payload) > 0 {
		bodyBytes, err := json.Marshal(payload)
		if err != nil {
			return nil, fmt.Errorf("error marshalling payload: %w", err)
		}

		body = bytes.NewReader(bodyBytes)
	}
```

Reply:
This worked, learned something new, thanks!